src/lib/i18n/index.ts

```diff
@@ -8,7 +8,12 @@ export const defaultLocale = locales[0];
 export const cookieName = 'HEAD_START_LOCALE';
 
 const i18n = rosetta(messages);
-i18n.locale(defaultLocale);
+
+if (typeof document !== 'undefined') {
+  i18n.locale(document.documentElement.lang as SiteLocale);
```

Comment:
Nice! I would use `setLocale` to protect against unsupported locales.

I'm also curious if we could add a test for this. Our `renderFragment` doesn't include the `document`, so we'll need a different method. Or maybe only mock just the `document` for this file?